services/blockchain-indexer/shared/database/mysqlKVStore.js

```diff
@@ -60,6 +53,29 @@ const get = async (key) => {
 	return value;
 };
 
+const get = async (key) => {
+	const keyValueDB = await getKeyValueStoreIndex();
+
+	const [{ value, type } = {}] = await keyValueDB.find(
+		{ key, limit: 1 },
+		['value', 'type'],
+	);
+
+	return formatValue(value, type);
+};
+
+const getByPattern = async (pattern) => {
+	const keyValueDB = await getKeyValueStoreIndex();
+
+	const result = await keyValueDB.find(
+		{ search: { property: 'key', pattern } },
+		['key', 'value', 'type'],
+	);
+
+	result.forEach(row => row.value = formatValue(row.value, row.type));
+	return result;
```

Comment:
```suggestion
	const formattedResult = result.map(row => {
		key: row.key,
		value: formatValue(row.value, row.type),
	});
	return formattedResult;
```

Reply:
this is creating another copy of the data whereas forEach will overwrite. since we are not doing anything else. would prefer keeping as is.

Reply:
I'm aware of the same. The reason for using map is that we do not want to send the `type` information in the response. That must be kept internal to the implementation.

Reply:
Applied.